platform/security/core/security-core-impl/src/main/java/ddf/security/samlp/impl/LogoutServiceImpl.java

```diff
@@ -0,0 +1,327 @@
+        Node node = (Node) XPathFactory.newInstance()
+                .newXPath()
+                .compile("//*[local-name(.) = 'Envelope']/*[local-name(.) = 'Body']")
+                .evaluate(responseDoc.getDocumentElement(), XPathConstants.NODE);
+        return DOM2Writer.nodeToString(node);
+    }
+
+    public String insertMsgIntoSoap(Element element) {
+        Envelope e = envelopeBuilder.buildObject();
+        Body b = bodyBuilder.buildObject();
+        b.setDOM(element);
+        e.setBody(b);
+        return DOM2Writer.nodeToString(e.getDOM());
+    }
+
+    public SignableSAMLObject extractXmlObject(String samlLogoutResponse)
+            throws WSSecurityException, XMLStreamException {
+        Document responseDoc =
+                StaxUtils.read(new ByteArrayInputStream(samlLogoutResponse.getBytes()));
+        XMLObject xmlObject = OpenSAMLUtil.fromDom(responseDoc.getDocumentElement());
+        if (xmlObject instanceof SignableSAMLObject) {
+            return (SignableSAMLObject) xmlObject;
+        }
+        return null;
+    }
+
+    private <T extends SAMLObject> T extract(String samlObject, Class<T> clazz)
+            throws WSSecurityException, XMLStreamException {
+        Document responseDoc = StaxUtils.read(new ByteArrayInputStream(samlObject.getBytes()));
+        XMLObject responseXmlObject = OpenSAMLUtil.fromDom(responseDoc.getDocumentElement());
+        if (clazz.isAssignableFrom(responseXmlObject.getClass())) {
+            return clazz.cast(responseXmlObject);
+        }
+        return null;
+    }
+
+    public LogoutResponse extractSamlLogoutResponse(String samlLogoutResponse)
+            throws XMLStreamException, WSSecurityException {
+        return extract(samlLogoutResponse, LogoutResponse.class);
+    }
+
+    public LogoutRequest extractSamlLogoutRequest(String samlLogoutResponse)
+            throws XMLStreamException, WSSecurityException {
+        return extract(samlLogoutResponse, LogoutRequest.class);
+    }
+
+    /**
+     * Returns a new <code>LogoutRequest</code> with a randomly generated ID, the current time
+     * for the <code>IssueInstant</code>, and <code>Version</code> set to <code>"2.0"</code>
+     *
+     * @param nameIdString     NameId of user to log out
+     * @param issuerOrEntityId The Issuer of the LogoutRequest
+     * @return the built <code>LogoutRequest</code>
+     */
+    public LogoutRequest buildLogoutRequest(String nameIdString, String issuerOrEntityId) {
+        return buildLogoutRequest(nameIdString,
+                issuerOrEntityId,
+                UUID.randomUUID()
+                        .toString());
+    }
+
+    public LogoutRequest buildLogoutRequest(String nameIdString, String issuerOrEntityId,
+            String id) {
+        LogoutRequest logoutRequest = logoutRequestBuilder.buildObject();
+
+        NameID nameID = nameIDBuilder.buildObject();
+        nameID.setValue(nameIdString);
```

Comment:
Yeah eventually I think we're gonna shift all the building stuff into SamlProtocol when we do the refactor